todoapp/TODOapp/app/src/main/java/com/nearsoft/androidschool/todoapp/activities/detail/DetailActivity.java

```diff
@@ -150,12 +198,12 @@ private void displayDetail() {
         dateSwitch.setChecked(todoItem.hasDate());
     }
 
-    private ToDoContent getTodo() throws AssertionError {
+    private ToDoContent getTodo() throws Exception {
         Bundle extras = getIntent().getExtras();
         if (extras != null && extras.containsKey(EXTRA_TODO_KEY)) {
             return (ToDoContent) extras.getSerializable(EXTRA_TODO_KEY);
         }
-        throw new AssertionError(getClass().getSimpleName() + " intent extras should contain an item");
+        throw new Exception(getClass().getSimpleName() + " intent extras should contain an item");
```

Comment:
@jmsalcido We no longer need the AssertionError because if the intent doesnt contain extras it means the user is creating a new ToDo